sources/packages/backend/apps/api/src/route-controllers/cas-invoice/tests/e2e/cas-invoice.aest.controller.updateInvoiceToResolved.e2e-spec.ts

```diff
@@ -0,0 +1,159 @@
+    app = nestApplication;
+    systemUsersService = app.get(SystemUsersService);
+    db = createE2EDataSources(dataSource);
+  });
+
+  it(`Should resolve an invoice in ${CASInvoiceStatus.ManualIntervention} status for the provided invoice identifier.`, async () => {
+    // Arrange
+    // Create CAS invoice batch.
+    const casInvoiceBatch = await db.casInvoiceBatch.save(
+      createFakeCASInvoiceBatch({
+        creator: systemUsersService.systemUser,
+      }),
+    );
+    // Creates part-time application with receipts, and invoices details.
+    const casInvoice = await saveFakeInvoiceIntoBatchWithInvoiceDetails(
+      db,
+      {
+        casInvoiceBatch,
+        creator: systemUsersService.systemUser,
+        // Part-time BC grants.
+        disbursementValues: [
+          createFakeDisbursementValue(
+            DisbursementValueType.BCGrant,
+            "BCAG",
+            200,
+            {
+              effectiveAmount: 150,
+            },
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.BCGrant,
+            "SBSD",
+            300,
+            {
+              effectiveAmount: 250,
+            },
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.BCTotalGrant,
+            "BCSG",
+            500,
+            { effectiveAmount: 400 },
+          ),
+        ],
+      },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        casSupplierInitialValues: {
+          supplierStatus: SupplierStatus.VerifiedManually,
+          supplierNumber: "222222",
+        },
+      },
+    );
+    casInvoice.invoiceStatus = CASInvoiceStatus.ManualIntervention;
+    await db.casInvoice.save(casInvoice);
+    const endpoint = `/aest/cas-invoice/${casInvoice.id}/resolve`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send()
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+    const updatedCASInvoice = await db.casInvoice.findOne({
+      select: {
+        id: true,
+        invoiceNumber: true,
+        invoiceStatus: true,
+        invoiceStatusUpdatedBy: { id: true },
+        invoiceStatusUpdatedOn: true,
+      },
+      relations: { invoiceStatusUpdatedBy: true },
+      where: { id: casInvoice.id },
+    });
+    const invoiceStatusUpdatedBy = await getAESTUser(
+      db.dataSource,
+      AESTGroups.BusinessAdministrators,
+    );
+    expect(updatedCASInvoice).toEqual({
+      id: casInvoice.id,
+      invoiceNumber: casInvoice.invoiceNumber,
+      invoiceStatus: CASInvoiceStatus.Resolved,
+      invoiceStatusUpdatedBy: { id: invoiceStatusUpdatedBy.id } as User,
+      invoiceStatusUpdatedOn: updatedCASInvoice.invoiceStatusUpdatedOn,
+    });
+  });
+
+  it("Should throw a HttpStatus Not Found (404) error when the CAS invoice to resolve doesn't exist.", async () => {
```

Comment:
Not a blocker again. But considering the importance of invoices which are in `Manual Intervention` status only  be resolved, test for that scenario will be nice to have. Again not a blocker.